core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -79,4 +80,15 @@ protected void deleteFiles(Set<String> pathsToDelete, String fileType) {
             (file, thrown) -> LOG.warn("Delete failed for {} file: {}", fileType, file, thrown))
         .run(deleteFunc::accept);
   }
+
+  protected Set<String> expiredStatisticsFilesLocations(
+      TableMetadata beforeExpiration, Set<Long> expiredIds) {
+    Set<String> expiredStatisticsFilesLocations = Sets.newHashSet();
+    for (StatisticsFile statisticsFile : beforeExpiration.statisticsFiles()) {
+      if (expiredIds.contains(statisticsFile.snapshotId())) {
+        expiredStatisticsFilesLocations.add(statisticsFile.path());
+      }
+    }
+    return expiredStatisticsFilesLocations;
+  }
```

Comment:
Curious, is it guaranteed that StatisticsFIles cannot be reused across multiple snapshots? 
The comment on StatisticsFile#snapshotId says 

`  /** ID of the Iceberg table's snapshot the statistics were computed from. */
  long snapshotId();
`
Is it possible that a StatisticsFile "file1" is computed at snapshot 1, but then still considered a valid reachable statistics file at Snapshot 2? and then we wouldn't want to do a removal of that file.

If it is possible it seems that we need to capture the reachability set rather than just removing all the statistics file before expiration. let me know if i misunderstood !

Reply:
How do we support time travel if we reuse the same statistics file for all the snapshots?
Because of this, I assumed that each snapshot will create its own statistics file.

Reply:
@amogh-jahagirdar good point
sorry, didn't read this comment before posting https://github.com/apache/iceberg/pull/6090/files#r1013923155
@ajantha-bhat see there for an example where you want to reuse a stats file

Reply:
@amogh-jahagirdar, @findepi  
As #6267 clarifies the spec. I have handled the reuse scenario and added the testcase now.